dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -108,6 +105,9 @@ def show
       return
     end
 
+    # will be true if the user is in any unarchived section where tts autoplay is enabled
+    @tts_autoplay_enabled = current_user&.sections_as_student&.where({hidden: false})&.map(&:tts_autoplay_enabled)&.reduce(false, :|)
```

Comment:
Oh I just moved that down below the redirect so that we wouldn't do an unnecessary query just to redirect afterwards! It was just a little optimization I noticed when I was looking through the code.